src/main/java/com/acquia/http/HMACMessageCreator.java

```diff
@@ -26,128 +28,267 @@
  *
  */
 public class HMACMessageCreator {
-    
+
+    private static final String ENCODING_UTF_8 = "UTF-8";
+
+    private static final String PARAMETER_AUTHORIZATION = "Authorization";
+    private static final String PARAMETER_X_AUTHORIZATION_TIMESTAMP = "X-Authorization-Timestamp";
+    private static final String PARAMETER_CONTENT_TYPE = "Content-Type";
+
+    /**
+     * Constructor
+     */
+    public HMACMessageCreator() {
+        super();
+    }
+
     /**
      * Create the message based on the given HTTP request received and list of custom headers.
      * 
      * @param request HTTP request
-     * @param customHeaders List of custom header names
      * @return The message to be encrypted
-     * @throws IOException If the body can not be read or the MD5 of the body can not be created
+     * @throws IOException if bodyHash cannot be created
      */
-    public String createMessage(HttpServletRequest request, List<String> customHeaders) throws IOException {
-        String httpVerb = request.getMethod();
+    public String createMessage(HttpServletRequest request) throws IOException {
+        String httpVerb = request.getMethod().toUpperCase();
+
+        int port = request.getServerPort();
+        String host = request.getServerName() + (port > 0 ? ":" + port : "");
+        String path = request.getRequestURI();
+        String queryParameters = request.getQueryString();
+        if (queryParameters == null) {
+            queryParameters = "";
+        }
+
+        String authorization = request.getHeader(PARAMETER_AUTHORIZATION);
+        HMACAuthorizationHeader authHeader = HMACAuthorizationHeader.getAuthorizationHeaderObject(authorization);
+
+        Map<String, String> authorizationHeaderParameterMap = authHeader.getEssentialHeaderMap();
```

Comment:
Can you get rid of this map and just explicitly build it in this method with the authHeader getters?